### PR TITLE
Reset logger in test to avoid global state persisting.

### PR DESCRIPTION
**Pull request by phiggins**
Signed-off-by: Pete Higgins <pete@peterhiggins.org>

Running these two tests together on Windows generates lots of warnings about rspec doubles persisting between tests:

```
PS C:\vagrant\chef> rspec .\spec\unit\log\winevt_spec.rb .\spec\unit\lwrp_spec.rb
<internal:warning>:43:in `warn': opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/cheffish-16.0.2/lib/cheffish/rspec/repository_support.rb:12
...and you are now defining it at:
  //vboxsvr/vagrant/chef/spec/support/shared/integration/integration_helper.rb:109
The new definition will overwrite the original one. (StructuredWarnings::BuiltInWarning)
Run options:
  include {:focus=>true}
  exclude {:provider=>#<Proc: //vboxsvr/vagrant/chef/spec/spec_helper.rb:212>, :arch=>#<Proc: //vboxsvr/vagrant/chef/spec/spec_helper.rb:206>, :requires_ifconfig=>true, :ruby=>"2.7.1", :chef=>"16.3.13", :not_intel_64bit=>true, :rhel_gte_8=>true, :rhel8=>true, :rhel7=>true, :rhel6=>true, :rhel=>true, :not_wpar=>true, :broken=>true, :openssl_lt_101=>true, :requires_root=>true, :selinux_only=>true, :debian_family_only=>true, :opensuse=>true, :suse_only=>true, :aix_only=>true, :linux_only=>true, :unix_only=>true, :solaris_only=>true, :windows_service_requires_assign_token=>true, :windows_domain_joined_only=>true, :windows_powershell_no_dsc_only=>true, :ruby32_only=>true, :windows_lt_10=>true, :windows32_only=>true, :win2012r2_only=>true, :macos_1014=>true, :macos_only=>true, :not_supported_on_windows=>true, :volatile_from_verify=>false, :volatile=>true, :external=>true}

All examples were filtered out; ignoring {:focus=>true}
......................................C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> received unexpected message :report_event with ({:data=>["Processing lwrp_foo[morpheus] action pass_buck (dynamically defined)"], :event_id=>10100, :event_type=>4, :source=>"chef"}) (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> received unexpected message :report_event with ({:data=>["Processing lwrp_foo[prepared_thumbs] action prepare_thumbs (C:/vagrant/chef/spec/data/lwrp/providers/buck_passer.rb line 14)"], :event_id=>10100, :event_type=>4, :source=>"chef"}) (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> received unexpected message :report_event with ({:data=>["Processing lwrp_foo[twiddled_thumbs] action twiddle_thumbs (C:/vagrant/chef/spec/data/lwrp/providers/buck_passer.rb line 21)"], :event_id=>10100, :event_type=>4, :source=>"chef"}) (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> received unexpected message :report_event with ({:data=>["Processing zen_master[keanu reeves] action nothing (dynamically defined)"], :event_id=>10100, :event_type=>4, :source=>"chef"}) (StructuredWarnings::StandardWarning)
.C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
........C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
.C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
.C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
.C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
.........C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/mixlib-log-3.0.8/lib/mixlib/log/logger.rb:78:in `rescue in write': log writing failed. #<InstanceDouble(Win32::EventLog) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for. (StructuredWarnings::StandardWarning)
.

Finished in 9.24 seconds (files took 17.58 seconds to load)
60 examples, 0 failures

PS C:\vagrant\chef>
```

There are other tests that change the logger and don't attempt any cleanup (https://github.com/chef/chef/blob/b6d4990724237f58d0c6beb17eaf2268c894579e/spec/unit/mixin/deprecation_spec.rb#L27), so resetting the logger state before each test runs seemed like the safer way to prevent this rather than copying and pasting setup and teardown code and expecting each test do the right thing.

### CODE REVIEW

**Comment by phiggins** [comment]
This behavior of swallowing any exception and emitting a warning seems like it could end in a scenario where you can't write to the logger and don't know why:

https://github.com/chef/mixlib-log/blob/d3a881afd49657e44c69b2f0f6bb1fa5edd52f01/lib/mixlib/log/logger.rb#L75-L79

(I'm guessing this is using `Kernel#warn` and not the logger's `warn` because that seems like a way more obvious bad situation.)

File: lib/chef/log.rb [hunk 1]
```diff
@@ -28,8 +28,13 @@ class Chef
...
     extend Mixlib::Log
 
+    def self.use_default_settings!
+      init(MonoLogger.new(STDOUT))
+      nil
+    end
+
     # Force initialization of the primary log device (@logger)
-    init(MonoLogger.new(STDOUT))
+    use_default_settings!
```
Comment (lamont-granquist) [inline lib/chef/log.rb:37]: this should probably just be `reset!` or something like that.

Comment (phiggins) [reply lib/chef/log.rb:37]: This already has a `reset!` method from `Mixlib::Log`. As far as I can tell, `init(MonoLogger.new(STDOUT))` is the same as calling `init` with no arguments, so a plain `reset!` would probably work, but I didn't want to introduce any weird side effects.

File: spec/unit/provider/execute_spec.rb [hunk 2]
```diff
@@ -43,17 +43,10 @@
...
     allow(ChefUtils).to receive(:windows?) { false }
-    @original_log_level = Chef::Log.level
     Chef::Log.level = :info
     allow(STDOUT).to receive(:tty?).and_return(false)
   end
 
-  after do
-    Chef::Log.level = @original_log_level
-    Chef::Config[:always_stream_execute] = false
-    Chef::Config[:daemon] = false
```
Comment (phiggins) [inline spec/unit/provider/execute_spec.rb:54]: `Chef::Config` is reset before each test so these lines weren't necessary.

File: lib/chef/log.rb (hunk 1 above)
Comment (lamont-granquist) [reply lib/chef/log.rb:37]: what about calling it `setup!`? (and i'm only picking a nit over the name, didn't realize that there was already a `reset!` method)

**Pull request merged**